### DISCUSSION TITLE
Why does " schema: "public" " gives error, while " schema: "extensions" " works?

### DISCUSSION THREAD

**Discussion by 03hgryan** [prisma/prisma#23336] [Q&A] [answered]
### Question

I am creating pgvector database, and

```
datasource db {
  provider   = "postgresql"
  url        = env("DATABASE_URL")
  extensions = [pgvector(map: "vector", schema: "extensions")]
}

datasource db {
  provider   = "postgresql"
  url        = env("DATABASE_URL")
  extensions = [vector]
}
```

worked, but

```
datasource db {
  provider   = "postgresql"
  url        = env("DATABASE_URL")
  extensions = [pgvector(map: "vector", schema: "public")]
}
```

this gave me ERROR: must be owner of extension vector.

Can anyone help me understand the issue, and explain me the differences between extensions = [pgvector(map: "vector", schema: "extensions")] and extensions = [vector] ?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @03hgryan 👋 

Thank you for raising this question.

The `pgvector(map: "vector", schema: "extensions")` and `vector` are both related to the `pgvector` extension, but they are used in slightly different ways. See this section of the [documentation](https://www.prisma.io/docs/orm/prisma-schema/postgresql-extensions#how-to-represent-postgresql-extensions-in-your-prisma-schema)

1. `pgvector(map: "vector", schema: "extensions")`: This is used to enable the pgvector extension and map it to the vector type in a specific schema, in this case, the extensions schema. The extensions schema must exist in your PostgreSQL database, and the database user must have the necessary permissions to create extensions in this schema.

2. `vector`: This is a simplified way to enable the pgvector extension in the default (public) schema. It's equivalent to pgvector(map: "vector", schema: "public"). You can use the simplified `extensions = [vector]` syntax, which will enable the `pgvector` extension in the default schema without requiring specific permissions.

The error message `ERROR: must be owner of extension vector` suggests that the database user specified in your `DATABASE_URL` does not have the necessary permissions to create or alter the `vector` extension in the `public` schema. You might need to grant these permissions to your database user or use a database user that already has these permissions.


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by 03hgryan** [reply]
Thanks! 🙏 🙏
